File: Components/CadDll/Demos/View/DemoC#/XmlExportParams.cs
using System;
using System.Text;
using System.Xml;
using System.Xml.Serialization;
using System.IO;
using System.Drawing;
using CADDLL;

namespace CADImageDemoCS
{
    [Serializable]
    public class ExportParams
    {
        [XmlAttribute]
        public string Version = "1.0";
        public string FileName;
        public string Ext;
        public CADParams CADParametrs = new CADParams();
        public GraphicParams GraphicParametrs = new GraphicParams();

        public override string ToString()
        {
            string rez = "";            
            StringBuilder sb = new StringBuilder();
            try
            {
                TextWriter wr = new StringWriter(sb);
                try
                {
                    XmlSerializer serializer = new XmlSerializer(typeof(ExportParams));
                    try
                    {
                        serializer.Serialize(wr, this);
                    }
                    finally
                    {
                        serializer = null;
                    }
                }
                finally
                {
                    wr.Close();
                    wr = null;
                }
                rez = sb.ToString();
            }
            finally
            {                    
                sb = null;
            }                
            return rez;
        }
    }

    [Serializable]
    public class CADParams
    {
        public int BackgroundColor = ColorTranslator.ToWin32(Color.White);
        public int DefaultColor = ColorTranslator.ToWin32(Color.Black);
        public Double XScale = 1.0;

        [XmlIgnore]
        public Color ColorBackground
        {
            get { return ColorTranslator.FromWin32(BackgroundColor); }
            set { BackgroundColor = ColorTranslator.ToWin32(value); }
        }

        [XmlIgnore]
        public Color ColorDefault
        {
            get { return ColorTranslator.FromWin32(DefaultColor); }
            set { DefaultColor = ColorTranslator.ToWin32(value); }
        }
    }

    [Serializable]
    public class GraphicParams
    {
        public int PixelFormat = 6;
        public int Width = 800;
        public int Height = 600;
        public int DrawMode = 0;
        public Rect DrawRect = new Rect();
    }
}

File: Components/CadDll/Demos/View/DemoVB.NET/XmlExportParams.vb
Imports System.Text
Imports System.Xml.Serialization
Imports System.IO

<Serializable()> Public Class ExportParams

    <XmlAttribute()> Public Version As String = "1.0"
    Public FileName As String
    Public Ext As String
    Public CADParametrs As CADParams = New CADParams()
    Public GraphicParametrs As GraphicParams = New GraphicParams()

    Public Overrides Function ToString() As String
        Dim rez As String = ""
        Dim sb As StringBuilder = New StringBuilder()
        Try
            Dim wr As TextWriter = New StringWriter(sb)
            Try
                Dim serializer As XmlSerializer = New XmlSerializer(GetType(ExportParams))
                Try
                    serializer.Serialize(wr, Me)
                Finally
                    serializer = Nothing
                End Try
            Finally
                wr.Close()
                wr = Nothing
            End Try
            rez = sb.ToString()
        Finally
            sb = Nothing
        End Try
        Return rez
    End Function

    <Serializable()> Public Class CADParams
        Public BackgroundColor As Integer = ColorTranslator.ToWin32(Color.White)
        Public DefaultColor As Integer = ColorTranslator.ToWin32(Color.Black)
        Public XScale As Double = 1.0

        <XmlIgnore()> Public Property ColorBackground() As Color
            Get
                Return ColorTranslator.FromWin32(BackgroundColor)
            End Get
            Set(ByVal value As Color)
                BackgroundColor = ColorTranslator.ToWin32(value)
            End Set
        End Property

        <XmlIgnore()> Public Property ColorDefault() As Color
            Get
                Return ColorTranslator.FromWin32(DefaultColor)
            End Get
            Set(ByVal value As Color)
                DefaultColor = ColorTranslator.ToWin32(value)
            End Set
        End Property
    End Class

    <Serializable()> Public Class GraphicParams
        Public PixelFormat As Integer = 6
        Public Width As Integer = 800
        Public Height As Integer = 600
        Public DrawMode As Integer = 0
        Public DrawRect As Rect = New Rect()
    End Class
End Class
